tests/unit/ListShareClass.spec.ts

```diff
@@ -0,0 +1,156 @@
+      hasRightsOrRestrictions: true,
+      series: [
+        {
+          id: 1,
+          name: 'Share Series 1',
+          priority: 1,
+          hasMaximumShares: true,
+          maxNumberOfShares: 50,
+          hasRightsOrRestrictions: false
+        },
+        {
+          id: 2,
+          name: 'Share Series 2',
+          priority: 2,
+          hasMaximumShares: true,
+          maxNumberOfShares: 100,
+          hasRightsOrRestrictions: false
+        }
+      ]
+    },
+    {
+      id: 2,
+      name: 'Non-voting Shares',
+      priority: 1,
+      maxNumberOfShares: 1000,
+      parValue: null,
+      currency: '',
+      hasRightsOrRestrictions: false,
+      series: [
+        {
+          id: 1,
+          name: 'Share Series 3',
+          priority: 1,
+          hasMaximumShares: true,
+          maxNumberOfShares: 50,
+          hasRightsOrRestrictions: false
+        }
+      ]
+    },
+    {
+      id: 3,
+      name: 'Common Shares 2',
+      priority: 2,
+      maxNumberOfShares: 10000,
+      parValue: 1.00,
+      currency: 'Canadian Dollar ($CDN)',
+      hasRightsOrRestrictions: true,
+      series: []
+    },
+    {
+      id: 4,
+      priority: 3,
+      name: 'Non-voting Shares 2',
+      maxNumberOfShares: 1000,
+      parValue: null,
+      currency: '',
+      hasRightsOrRestrictions: false,
+      series: []
+    }]
+
+  beforeEach(() => {
+    const localVue = createLocalVue()
+    vuetify = new Vuetify({})
+
+    wrapper = mount(ListShareClass, {
+      localVue,
+      vuetify,
+      propsData: {
+        shareClasses: shareClasses
+      }
+    })
+  })
+
+  afterEach(async () => {
+    await wrapper.destroy()
+  })
+
+  it('displays the correct amount of share classes / series when data is present', () => {
+    const classRowCount = wrapper.vm.$el.querySelectorAll('.v-data-table .class-row').length
+    const seriesRowCount = wrapper.vm.$el.querySelectorAll('.v-data-table .series-row').length
+
+    expect(classRowCount).toEqual(4)
+    expect(seriesRowCount).toEqual(3)
+  })
+
+  it('displays the correct name data in the share classes / series table', () => {
+    const classListItem1 = wrapper.vm.$el.querySelectorAll('.v-data-table .class-row')[0]
+    const seriesListItem1 = wrapper.vm.$el.querySelectorAll('.v-data-table .series-row')[0]
+
+    expect(classListItem1.querySelector('.list-item__title').textContent).toContain('Common Shares')
+    expect(seriesListItem1.querySelector('.series-name').textContent).toContain('Share Series 1')
+  })
+
+  it('displays the correct data in the selected table rows for shareClass', () => {
+    const classListItem1 = wrapper.vm.$el.querySelectorAll('.v-data-table .class-row')[2]
+
+    expect(classListItem1.querySelectorAll('td')[0].textContent).toContain('Common Shares 2')
+    expect(classListItem1.querySelectorAll('td')[1].textContent).toContain(10000)
+    expect(classListItem1.querySelectorAll('td')[2].textContent).toContain('$1')
```

Comment:
Might as well check for longest match here, eg, "$1.00".

Reply:
I changed data set to give a better representation of possible values